sycl/test/basic_tests/implicit_conversion_error.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsyntax-only -Xclang -verify  %s -I %sycl_include -Xclang -verify-ignore-unexpected=note,warning
```

Comment:
Do we really need to change this test as it is compile-time check?

Reply:
The new static_assert at stl.hpp is emitted on Windows when a) user does not use -fsycl and b) when such compilation would potentially lead to linking with static RT.
I will check if there is any difference in compilation macros with/with -fsyntax-only and if that difference usable in stl.hpp to not give an error with -fsyntax-only.